internal/controller/auth_controller.go

```diff
@@ -47,30 +43,67 @@ func NewAuthController(userRepo services.UserService, loginAttemptsService servi
 // @Failure      409  {object}  utils.HTTPError "Email already exists"
 // @Failure      500  {object}  utils.HTTPError "Internal server error"
 // @Router       /auth/users [post]
-// @Router       /auth/admins [post]
-func (ac *AuthController) Register(admin bool) gin.HandlerFunc {
-	return func(c *gin.Context) {
-		var request models.CreateUserRequest
-		ctx := c.Request.Context()
-		if err := c.Bind(&request); err != nil {
-			utils.ErrorResponse(c, http.StatusBadRequest, "Invalid request format")
-			return
-		}
+func (ac *AuthController) Register(c *gin.Context) {
+	var request models.CreateUserRequest
+	ctx := c.Request.Context()
+	if err := c.ShouldBindJSON(&request); err != nil {
+		utils.ErrorResponse(c, http.StatusBadRequest, "Invalid request format")
+		return
+	}
 
-		if _, err := ac.userRepo.GetUserByEmail(ctx, request.Email); err == nil {
-			utils.ErrorResponse(c, http.StatusConflict, "Email already exists")
-			return
-		}
+	if _, err := ac.userRepo.GetUserByEmail(ctx, request.Email); err == nil {
+		utils.ErrorResponse(c, http.StatusConflict, "Email already exists")
+		return
+	}
```

Comment:
```suggestion
	if _, err := ac.userRepo.GetUserByEmail(ctx, request.Email); err == nil {
		utils.ErrorResponse(c, http.StatusConflict, "Email already exists")
		return
	}
```
	
	esto deberia chequear si el user esta verificado on no, si no esta verificado borra el user y funciona como si estuviera registrandose de nuevo, si no no tiene forma de entrar devuelta a la pantalla para poner el PIN, sino devolver un error en especifico para hacerlo saber al front que el user todavia no esta verificado asi reenvia el pin, pero creo que es mejor si lo elimina (en el caso de que se quiera poner otra contraaseña o algo)